src/main/com/yetanalytics/lrs/pedestal/routes.cljc

```diff
@@ -39,29 +42,35 @@
         method-not-allowed
         :route-name (keyword route-name-ns (name method))]
        (let [params-interceptors
-             [(xapi-i/params-interceptor
-               (case resource-tuple
-                 ["activities" "state"]
-                 (case method
-                   :put    :xapi.activities.state.PUT.request/params
-                   :post   :xapi.activities.state.POST.request/params
-                   :get    :xapi.activities.state.GET.request/params
-                   :head   :xapi.activities.state.GET.request/params
-                   :delete :xapi.activities.state.DELETE.request/params)
-                 ["activities" "profile"]
-                 (case method
-                   :put    :xapi.activities.profile.PUT.request/params
-                   :post   :xapi.activities.profile.POST.request/params
-                   :get    :xapi.activities.profile.GET.request/params
-                   :head   :xapi.activities.profile.GET.request/params
-                   :delete :xapi.activities.profile.DELETE.request/params)
-                 ["agents"     "profile"]
-                 (case method
-                   :put    :xapi.agents.profile.PUT.request/params
-                   :post   :xapi.agents.profile.POST.request/params
-                   :get    :xapi.agents.profile.GET.request/params
-                   :head   :xapi.agents.profile.GET.request/params
-                   :delete :xapi.agents.profile.DELETE.request/params)))]
+             (cond->
+                 [(xapi-i/params-interceptor
+                   (case resource-tuple
+                     ["activities" "state"]
+                     (case method
+                       :put    :xapi.activities.state.PUT.request/params
+                       :post   :xapi.activities.state.POST.request/params
+                       :get    :xapi.activities.state.GET.request/params
+                       :head   :xapi.activities.state.GET.request/params
+                       :delete :xapi.activities.state.DELETE.request/params)
+                     ["activities" "profile"]
+                     (case method
+                       :put    :xapi.activities.profile.PUT.request/params
+                       :post   :xapi.activities.profile.POST.request/params
+                       :get    :xapi.activities.profile.GET.request/params
+                       :head   :xapi.activities.profile.GET.request/params
+                       :delete :xapi.activities.profile.DELETE.request/params)
+                     ["agents"     "profile"]
+                     (case method
+                       :put    :xapi.agents.profile.PUT.request/params
+                       :post   :xapi.agents.profile.POST.request/params
+                       :get    :xapi.agents.profile.GET.request/params
+                       :head   :xapi.agents.profile.GET.request/params
+                       :delete :xapi.agents.profile.DELETE.request/params)))]
```

Comment:
This big ol' `case` expression should be extracted out now that we're wrapping a `cond->` around it.

Reply:
aad95fc3f2834ddb6da17ce0cc9b13fe32f1f5d3

Reply:
97f9cd1df1f96f0645dc6b13a19e46cee47acbbb